### PR TITLE
[Merged by Bors] - Add 0.10 news about "stageless" Schedulev3 

### PR DESCRIPTION
**Pull request by alice-i-cecile**
To do:

- [x] fix all lints
- [x] ensure everyone is credited
- [x] fix all links
- [x] add schedule graph image

### CODE REVIEW

File: content/news/2023-03-04-bevy-0.10/index.md [hunk 1]
```diff
@@ -15,17 +15,411 @@ Since our last release a few months ago we've added a _ton_ of new features, bug
...
+```
+
+Pretty simple, but what does this buy us?
+First, it gives you a clear hook to impose, reason about and visualize high level structure to your schedule. Yearning for a linear, stage-like design? Just order your base sets!
+Secondly, it allows Bevy to set good default behavior for systems added by users, without removing their control.
+
+Let me tell you a story, set in a world where all of Mr. Straw Man's points above are true, and no default set is added.
+
+1. A new user adds the `make_player_run` system to their app.
+2. Sometimes this system runs before input handling, leading to randomly dropped inputs. Sometimes it runs after rendering, leading to stranges flickers.
```
Comment (LiamGallagher737) [inline content/news/2023-03-04-bevy-0.10/index.md:319]:
```suggestion
2. Sometimes this system runs before input handling, leading to randomly dropped inputs. Sometimes it runs after rendering, leading to strange flickers.
```

File: content/news/2023-03-04-bevy-0.10/index.md [hunk 2]
```diff
@@ -15,17 +15,411 @@ Since our last release a few months ago we've added a _ton_ of new features, bug
...
+#[derive(SystemSet, Debug, Hash, PartialEq, Eq, Clone)]
+enum PhysicsSet {
+    Forces,
+    Kinematics,
+    CollisionDetection
+}
+
+app
+    // .with_run_criteria -> .run_if <3
+   .add_system(gravity.in_set(PhysicsSet::Forces).run_if(gravity_enabled))
```
Comment (theotherphil) [inline content/news/2023-03-04-bevy-0.10/index.md:64]: Comprehension check from a casual reader: will this run condition apply to any systems added to this set or have I misunderstood the comment above about run conditions and shared configuration?

File: content/news/2023-03-04-bevy-0.10/index.md [hunk 3]
```diff
@@ -15,17 +15,411 @@ Since our last release a few months ago we've added a _ton_ of new features, bug
...
+Run criteria have been renamed to the clearer **run conditions**, which can be constructed out of any read-only system that returns `bool`.
+
+With a new blessed pattern for complex control flow, we can finally get rid of looping run criteria. [`ShouldRun::YesAndCheckAgain`](https://docs.rs/bevy/0.9.1/bevy/ecs/schedule/enum.ShouldRun.html) was not exactly straightforward to reason about, either for engine devs or users. It's always a bad sign when your bool-like enums have four possible values.
+
+If you crave that powerful, complex control flow: use the "schedules in exclusive systems" pattern listed above.
+For the other 99% of use cases, enjoy the simpler `bool`-based run conditions.
+
+```rust!
+// Let's make our own run condition
+fn contrived_run_condition(query: Query<&Life, With<Player>>, score: Res<Score>) -> bool{
```
Comment (theotherphil) [inline content/news/2023-03-04-bevy-0.10/index.md:191]: Nit: missing a space before the bracket at the end of the line

Comment (alice-i-cecile) [reply content/news/2023-03-04-bevy-0.10/index.md:191]:
```suggestion
fn contrived_run_condition(query: Query<&Life, With<Player>>, score: Res<Score>) -> bool {
```

File: content/news/2023-03-04-bevy-0.10/index.md [hunk 4]
```diff
@@ -15,17 +15,411 @@ Since our last release a few months ago we've added a _ton_ of new features, bug
...
+// Each variant of this enum is a distinct system set.
+#[derive(SystemSet, Debug, Hash, PartialEq, Eq, Clone)]
+enum PhysicsSet {
+    Forces,
+    Kinematics,
+    CollisionDetection
+}
+
+app
+    // .with_run_criteria -> .run_if <3
```
Comment (alice-i-cecile) [inline content/news/2023-03-04-bevy-0.10/index.md:63]:
```suggestion
    // .with_run_criteria -> .run_if <3
    // Note that in this case we're modifying this single `gravity` system, not the entire `PhysicsSet::Forces`
    // as this is a method on a single system.
    // The order of these method calls doesn't matter!
```

Comment (alice-i-cecile) [reply content/news/2023-03-04-bevy-0.10/index.md:63]: @theotherphil is this helpful to you?

Comment (theotherphil) [reply content/news/2023-03-04-bevy-0.10/index.md:63]: Yes, although the thing that confused me was the comment about sharing whether-there-are-run-conditions in the previous paragraph. Without that I would have assumed that this condition was specific to the system.

File: content/news/2023-03-04-bevy-0.10/index.md [hunk 5]
```diff
@@ -15,17 +15,411 @@ Since our last release a few months ago we've added a _ton_ of new features, bug
...
+
+[!main_schedule_diagram](main_schedule_diagram.svg)
+
+This diagram, made with [@jakobhellermann's `bevy_mod_debugdump` crate](https://github.com/jakobhellermann/bevy_mod_debugdump) shows a simplified version of Bevy's default schedule.
+
+## Configurable System Sets
+
+To support more natural and flexible control over "how are my systems run and scheduled", the idea of a "system set" has been redefined, rolling up the existing "system label" concept into one straightforward but powerful abstraction.
+
+**System sets** are named collections of systems that share a set of **system configuration**: if there are run conditions attached, how they are ordered relative to other systems or sets and so on. This is distributive: Ordering systems relative to a system set applies that ordering to _all_ systems in that set.
```
Comment (alice-i-cecile) [inline content/news/2023-03-04-bevy-0.10/index.md:46]:
```suggestion
Every system, once it is part of a schedule, stores **system configuration** metadata: f there are run conditions attached, how they are ordered relative to other systems or sets and so on.
**System sets** are named collections of systems that share system configuration across all of their members. This is both distributive and additive: ordering systems relative to a system set applies that ordering to _all_ systems in that set, in addition to any configuration on each individual system.
```

File: content/news/2023-03-04-bevy-0.10/index.md (hunk 4 above)
Comment (alice-i-cecile) [reply content/news/2023-03-04-bevy-0.10/index.md:64]: Got it!

**Comment by theotherphil** [comment]
Somewhat. I’m still not clear what it means for “the existence of run conditions” to be shared across the systems within a set, but I don’t want my confusion to take away too much of your time! Thanks for your work on the feature and write up - there’s always a load of cool new stuff to read about in every release.

File: content/news/2023-03-04-bevy-0.10/index.md [hunk 6]
```diff
@@ -15,17 +15,417 @@ Since our last release a few months ago we've added a _ton_ of new features, bug
...
+
+[!main_schedule_diagram](main_schedule_diagram.svg)
+
+This diagram, made with [@jakobhellermann's `bevy_mod_debugdump` crate](https://github.com/jakobhellermann/bevy_mod_debugdump) shows a simplified version of Bevy's default schedule.
+
+## Configurable System Sets
+
+To support more natural and flexible control over "how are my systems run and scheduled", the idea of a "system set" has been redefined, rolling up the existing "system label" concept into one straightforward but powerful abstraction.
+
+Every system, once it is part of a schedule, stores **system configuration** metadata: f there are run conditions attached, how they are ordered relative to other systems or sets and so on.
```
Comment (alice-i-cecile) [inline content/news/2023-03-04-bevy-0.10/index.md:46]:
```suggestion
Every system, once it is part of a schedule, stores **system configuration** metadata: which run conditions are attached, how they are ordered relative to other systems or sets and so on.
```

File: content/news/2023-03-04-bevy-0.10/index.md [hunk 7]
```diff
@@ -15,17 +15,417 @@ Since our last release a few months ago we've added a _ton_ of new features, bug
...
+```
+
+Pretty simple, but what does this buy us?
+First, it gives you a clear hook to impose, reason about and visualize high level structure to your schedule. Yearning for a linear, stage-like design? Just order your base sets!
+Secondly, it allows Bevy to set good default behavior for systems added by users, without removing their control.
+
+Let me tell you a story, set in a world where all of Mr. Straw Man's points above are true, and no default set is added.
+
+1. A new user adds the `make_player_run` system to their app.
+2. Sometimes this system runs before input handling, leading to randomly dropped inputs. Sometimes it runs after rendering, leading to stranges flickers.
```
Comment (alice-i-cecile) [inline content/news/2023-03-04-bevy-0.10/index.md:323]:
```suggestion
2. Sometimes this system runs before input handling, leading to randomly dropped inputs. Sometimes it runs after rendering, leading to strange flickers.
```

File: content/news/2023-03-04-bevy-0.10/index.md [hunk 8]
```diff
@@ -0,0 +1,442 @@
...
+* **Pull `!Send` data out of the `World`:** storing non thread-safe data in a structure designed to be sent across threads has caused us no end of headaches. We plan on pulling these out into the `App`, resolving a major blocker for a first-class [multiple worlds](https://github.com/bevyengine/rfcs/pull/43) design.
+* **Timestamp window and input events:** As discussed in [#5984](https://github.com/bevyengine/bevy/issues/5984), tracking the exact timing of input events is essential to ensuring that event ordering and timing can be precisely reconstructed.
+* **[Opt-out change detection](https://github.com/bevyengine/bevy/issues/4882):** improve performance for tiny components by turning off change detection at compile or run-time.
+* **[Batched queries](https://github.com/bevyengine/bevy/pull/6161):** explicit SIMD-accelerated query iteration to get the most out of modern CPUs.
```
Comment (alice-i-cecile) [inline content/news/2023-03-04-bevy-0.10/index.md:428]:
```suggestion
```

File: content/news/2023-03-04-bevy-0.10/index.md [hunk 9]
```diff
@@ -15,17 +15,417 @@ Since our last release a few months ago we've added a _ton_ of new features, bug
...
 
-<div class="release-feature-authors">authors: @Foo, @Bar</div>
+<div class="release-feature-authors">authors: @alice-i-cecile, @maniwani, @WrongShoe, @jakobhellermann, @JoJoJet, @geieredgar and a whole lot more </div>
 
-Description here.
+Thanks to the fantastic work of our ECS team, the hotly awaited ["stageless" scheduling RFC](https://github.com/bevyengine/rfcs/blob/main/rfcs/45-stageless.md) has been implemented! But as we all know, plans and implementations (start at [#6587](https://github.com/bevyengine/bevy/pull/6587) by `@maniwani` and [#7267](https://github.com/bevyengine/bevy/pull/7267) by `@alice-i-cecile`) are two different things. Let's take a look at what actually shipped for 0.10.
+
+There's been a lot of changes, but we really do think that ripping off the band-aid now (before any form of stability guarantees) is essential to the health of Bevy's scheduling model going forward.
+
+The [migration path](../../learn/book/migration-guides/0.9-0.10/_index.md) for existing applications won't be trivial, but we've done our best to keep it surprisingly straightforward. Don't sweat it!
```
[the remark this answers is not in the document]
Comment (alice-i-cecile) [reply content/news/2023-03-04-bevy-0.10/index.md:28]: This was in request to feedback from a user who *had* migrated, and suggested that I make it scarier / show more empathy.

They were making use of the state stack though, so I think their experience was atypical.

That said, having done the migration for the engine itself I really didn't find it too bad at all.

File: content/news/2023-03-04-bevy-0.10/index.md [hunk 10]
```diff
@@ -15,17 +15,417 @@ Since our last release a few months ago we've added a _ton_ of new features, bug
...
+    // .with_run_criteria -> .run_if <3
+    // Note that in this case we're modifying this single `gravity` system, not the entire `PhysicsSet::Forces`
+    // as this is a method on a single system.
+    // The order of these method calls doesn't matter!
+   .add_system(gravity.in_set(PhysicsSet::Forces).run_if(gravity_enabled))
+    // Add multiple systems at once with add_systems!    
+    .add_systems((
+        apply_acceleration,
+        apply_velocity
+    // Quickly order a list of systems with .chain()
```
[the remark this answers is not in the document]
Comment (alice-i-cecile) [reply content/news/2023-03-04-bevy-0.10/index.md:73]:
```suggestion
    // Quickly order a list of systems to run one after the next by using .chain()
```

File: content/news/2023-03-04-bevy-0.10/index.md [hunk 11]
```diff
@@ -15,17 +15,417 @@ Since our last release a few months ago we've added a _ton_ of new features, bug
...
+
+These rules must be compatible with each other: any paradoxes (like a system set inside of itself, or a system that must run both before and after a set) will result in a runtime panic with a helpful error message.
+
+As long as you can construct the type of a system set, you can both order your systems relative to it, and configure its behavior even after it has been initialized elswhere! Crucially system configuration is strictly additive: you cannot _remove_ rules added elsewhere. This is both a "anti-spaghetti" and "plugin privacy" consideration. When this rule is combined with Rust's robust type privacy rules, plugin authors can make careful decisions about which exact invariants need to be upheld, and reorganize code and systems internally without breaking consumers.
+
+Plugin authors: consider offering both a "default configuration" and a "minimal configuration" version of your plugins to support more unusual scheduling patterns while keeping that "it just works" behavior you've come to love.
```
[the remark this answers is not in the document]
Comment (alice-i-cecile) [reply content/news/2023-03-04-bevy-0.10/index.md:102]: Yeah, this is a conversation for another day / place. I'm cutting this from this post.

File: content/news/2023-03-04-bevy-0.10/index.md [hunk 12]
```diff
@@ -0,0 +1,441 @@
...
```
Comment (alice-i-cecile) [inline content/news/2023-03-04-bevy-0.10/index.md:102]:
```suggestion
```

File: content/news/2023-03-04-bevy-0.10/index.md [hunk 13]
```diff
@@ -15,17 +15,417 @@ Since our last release a few months ago we've added a _ton_ of new features, bug
...
+Run criteria have been renamed to the clearer **run conditions**, which can be constructed out of any read-only system that returns `bool`.
+
+With a new blessed pattern for complex control flow, we can finally get rid of looping run criteria. [`ShouldRun::YesAndCheckAgain`](https://docs.rs/bevy/0.9.1/bevy/ecs/schedule/enum.ShouldRun.html) was not exactly straightforward to reason about, either for engine devs or users. It's always a bad sign when your bool-like enums have four possible values.
+
+If you crave that powerful, complex control flow: use the "schedules in exclusive systems" pattern listed above.
+For the other 99% of use cases, enjoy the simpler `bool`-based run conditions.
+
+```rust!
+// Let's make our own run condition
+fn contrived_run_condition(query: Query<&Life, With<Player>>, score: Res<Score>) -> bool {
```
[the remark this answers is not in the document]
Comment (alice-i-cecile) [reply content/news/2023-03-04-bevy-0.10/index.md:193]: Goal was to showcase that you could access both queries and resources.

Your suggestion is much better for that though!

File: content/news/2023-03-04-bevy-0.10/index.md [hunk 14]
```diff
@@ -15,17 +15,417 @@ Since our last release a few months ago we've added a _ton_ of new features, bug
...
+
+Bevy 0.10 is shipping with a lovely collection of built-in [common run conditions](https://dev-docs.bevyengine.org/bevy/ecs/schedule/common_conditions/index.html). Courtesy of [#6587 by `@maniwani`](https://github.com/bevyengine/bevy/pull/6587), [#7579 by `@inodentry`](https://github.com/bevyengine/bevy/pull/7579)and [#7806 by `@jakobhellermann`](https://github.com/bevyengine/bevy/pull/7806), you can quickly check if there are events to process, changes to resources, input states and more.
+
+When you need something more sophisticated, combining run conditions is a breeze. Courtesy of [#7547](https://github.com/bevyengine/bevy/pull/7547), [#7559](https://github.com/bevyengine/bevy/pull/7559), and [#7605](https://github.com/bevyengine/bevy/pull/7605), you can create new run conditions with the use of system piping and the `not`, `and_then` or `or_else` run criteria combinators.
+
+## Simpler States
+
+Of course, looping run criteria were used to power states.
```
[the remark this answers is not in the document]
Comment (alice-i-cecile) [reply content/news/2023-03-04-bevy-0.10/index.md:212]:
```suggestion
Previously, looping run criteria were used to power states, but as mentioned above, they've been removed.
```

File: content/news/2023-03-04-bevy-0.10/index.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (cart) [reply content/news/2023-03-04-bevy-0.10/index.md]: I deeply appreciate that you kept the "over 9000" reference :heart:

**Review by cart — APPROVED** [review]
I think you struck a nice balance between detailed and "easily grok-able". The tone is fun. I think this will get people appropriately excited!

**Pull request closed without merging**